src/engine/ult.c

```diff
@@ -310,59 +310,54 @@ sched_ult2xs(int xs_type, int tgt_id)
 	switch (xs_type) {
 	case DSS_XS_SELF:
 		return DSS_XS_SELF;
+	case DSS_XS_SYS:
+		return 0;
+	case DSS_XS_SWIM:
+		return 1;
+	case DSS_XS_DRPC:
+		return 2;
```

Comment:
this is fixed value as well, so I am not sure why commit msg with "otherwise the swim ULT will be still assigned in the same core(xstream) as the system xstream(0)."?

Reply:
reworked the logic.